### PR TITLE
Update to Python 3.7

### PR DESCRIPTION
**Pull request by ocefpaf**

### CODE REVIEW

**Comment by ccordoba12** [comment]
Please don't merge this until I have time to review it.

**Comment by ocefpaf** [comment]
It is a boilerplate update with virtually nothing really to review. Mostly changes that must happen to get this ready for the bot and the next generation of recipes.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,52 +1,51 @@
...
+    - pyflakes
+    - pygments >=2.0
     - pylint
-    - pyzmq
-    - chardet >=2.0
-    - numpydoc
-    - cloudpickle
-    - keyring  # [not (linux and py2k)]
-    - spyder-kernels =0
+    - pyqt >=5.6,<5.7
```
Comment (ccordoba12) [inline recipe/meta.yaml]: This is wrong. It should be `pyqt >=5.6,<6`.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,52 +1,51 @@
...
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  url: https://pypi.io/packages/source/s/spyder/spyder-{{ version }}.tar.gz
+  sha256: 992d74f07961df9a9b9aa495fa2d87832699ada5323850dc3e26ac0ff640b170
   patches:
     # Prevent a crash in macOS with Python 3
     - zmq.patch
```
Comment (ccordoba12) [inline recipe/meta.yaml]: I'll remove this patch. It's the same as the second one but just in a different place in the file.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,52 +1,49 @@
...
+    - pyflakes
+    - pygments >=2.0
     - pylint
-    - pyzmq
-    - chardet >=2.0
-    - numpydoc
-    - cloudpickle
-    - keyring  # [not (linux and py2k)]
-    - spyder-kernels =0
+    - pyqt >=5.6,<6
```
Comment (ocefpaf) [inline recipe/meta.yaml]: This is dangerous. The previous pin was set to avoid compatibility issues with `defaults`.

Comment (ccordoba12) [reply recipe/meta.yaml]: You can still install PyQt 5.6 from defaults, so how is that mix avoided?

Comment (ocefpaf) [reply recipe/meta.yaml]: Not b/c 5.6 from defaults is OK.

**Comment by ocefpaf** [comment]
Closing this to @ccordoba12 can do it the way he believes is better. The reason I was doing this was for a friend who used to use spyder and she mentioned she no longer needs this on py37.

**Comment by ccordoba12** [comment]
Ok, we'll wait for the regro bot then.

**Pull request closed without merging**